async/awaitを使うと数珠繋ぎでasync/awaitが連鎖しがち。
そんな時に非同期処理を同期として呼び出したい場合に以下の2パターンが多い様子。。例外で差分があるようなので比較した。AggregrateExceptinoとして捕捉。InnerException見る必要があるのがメンドクサイ。InvalidOperationExceptionとして捕捉。Wait()は良くある別タスク内で発生した例外同様AggreateExceptinoでまとめてChatchされる。
一方でGetAwaiter().GetResult()は同期メソッドで例外が発生したかの如く、発生時の状態で補足される。個人的にはGetAwaiter().GetResult()の方が好きだが別意見の人もいそう


